### PR TITLE
Wildfly 26 with jdk 17 base image #148

### PR DESCRIPTION
**Pull request by Henri-Laiho**

### CODE REVIEW

**Comment by jwgmeligmeyling** [comment]
Thanks for putting this Dockerfile out there 😄  Startup times went from 5 minutes to 13 seconds here. 🤣

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,9 @@
-# Use latest jboss/base-jdk:11 image as the base
-FROM jboss/base-jdk:11
+FROM eclipse-temurin:17.0.4.1_1-jdk-centos7
```
Comment (jmesnil) [inline Dockerfile:1]: If we do that switch, we will have to figure out how to name our images as I'd like to be able to provide JDK11 and 17 versions of the images.

We will likely have 1 a single Dockerfile and have the base image (either `eclipse-temurin:17-centos7` or `eclipse-temurin:11-centos7`) set as a parameter when the GitHub action builds the image.

The names of the WildFLy images would have to reflect this. With something like

`quay.io/wildfly/wildfly:26.1.2-jdk11`
`quay.io/wildfly/wildfly:26.1.2-jdk17`
`quay.io/wildfly/wildfly:latest` would point to the "oldest" supported JDK so `26.1.2-jdk11`.

wdyt?

Comment (Henri-Laiho) [reply Dockerfile:1]: I can add this in the beginning:
```
ARG jdk=11
FROM eclipse-temurin:${jdk}-centos7
```

This way 
`docker build .` gives openjdk 11.0.16.1
`docker build . --build-arg jdk=11` gives openjdk 11.0.16.1
`docker build . --build-arg jdk=17` gives openjdk 17.0.4.1
Would it be enough to set up the GitHub action?

Comment (jmesnil) [reply Dockerfile:1]: yes, that'd enough. thanks

**Comment by krisgerhard** [comment]
We have used this modification in production for about two weeks now. 🤘
@jmesnil Unfortunately OpenJDK is not equally valuable as the official support for arm based processors is still lacking.

This PR addresses a lot of existing issues and PRs. I'll list some of them.

https://github.com/jboss-dockerfiles/wildfly/pull/68
https://github.com/jboss-dockerfiles/wildfly/issues/155
https://github.com/jboss-dockerfiles/wildfly/issues/149
https://github.com/jboss-dockerfiles/wildfly/issues/148
https://github.com/jboss-dockerfiles/wildfly/issues/128
https://github.com/jboss-dockerfiles/wildfly/issues/108

**Comment by jmesnil** [comment]
@Henri-Laiho @krisgerhard I've followed up this PR with a change to build JDK 11 and 17 images using the GitHub action in #164 

I've pushed the images to my own container registry at https://quay.io/repository/jmesnil/wildfly?tab=tags.

Could you please have a look and run them on ARM (I don't have such hardware at the moment)?
If that's ok, we will be able to push them to the official `wildfly` registry.

**Review by jmesnil — CHANGES_REQUESTED** [review]
This looks good to me, thanks!

Could you squash all commits before I merge please?

**Comment by Henri-Laiho** [comment]
@jmesnil commits squashed

**Review by jmesnil — APPROVED** [review]

**Comment by krisgerhard** [comment]
> @Henri-Laiho @krisgerhard I've followed up this PR with a change to build JDK 11 and 17 images using the GitHub action in #164
> 
> I've pushed the images to my own container registry at https://quay.io/repository/jmesnil/wildfly?tab=tags.
> 
> Could you please have a look and run them on ARM (I don't have such hardware at the moment)? If that's ok, we will be able to push them to the official `wildfly` registry.

Verified `quay.io/jmesnil/wildfly:26.1.2.Final-1-jdk17` from ourside. Seems to work as expected on both AMD and ARM architectures. 

Thank you! 🤘

**Comment by jmesnil** [comment]
We're almost there. I've opened https://github.com/wildfly/wildfly.org/pull/369 to communicate on the changes and will merge the PRs soon...

**Comment by jmesnil** [comment]
@Henri-Laiho thanks!

**Pull request merged**